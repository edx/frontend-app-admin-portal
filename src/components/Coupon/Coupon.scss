@import "~bootstrap/scss/functions";
@import "~bootstrap/scss/variables";
@import "~bootstrap/scss/mixins";

.coupon {
    box-shadow: 0 .125rem .25rem rgba(0,0,0,0.075);
    transition: all 0.2s ease;

    &.dimmed {
        opacity: 0.5;

        &:hover,
        &:focus {
            opacity: 1;
        }
    }

    &:not(.expanded) {
        &:hover,
        &:focus {
            border-color: $gray-400 !important; // `!important` needed to override .border Bootstrap class
        }
    }

    .metadata {
        transition: background 0.2s ease;
        position: relative;

        &:hover,
        &:focus {
            background: $gray-100;
            cursor: pointer;
        }

        .toggle-details-btn {
            position: absolute;
            top: 0;
            right: 0;

            @include media-breakpoint-up(lg) {
                position: relative;
            }
        }
    }

    &.expanded {
        &:hover,
        &:focus {
            background: inherit;
        }

        .metadata {
            background: #095686;
            border-radius: 0.1875rem 0.1875rem 0 0;
            color: #ffffff;
        }
    }

    .details {
        font-size: 1rem;

        .loading {
            font-size: inherit;
        }

        .alert {
            margin-bottom: 0;
        }

        .table,
        .pagination .btn {
            font-size: 0.85rem;
        }

        .table {
            td {
                vertical-align: middle;
            }

            .btn {
                font-size: inherit;
                text-decoration: underline;
            }
        }
    }
}
