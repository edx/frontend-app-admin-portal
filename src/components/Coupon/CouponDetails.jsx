import React from 'react';
import { Button, CheckBox, Icon } from '@edx/paragon';

import H3 from '../H3';

import TableContainer from '../../containers/TableContainer';
import DownloadCsvButton from '../../containers/DownloadCsvButton';
import EcommerceApiService from '../../data/services/EcommerceApiService';

class CouponDetails extends React.Component {
  constructor(props) {
    super(props);

    this.tableColumns = [
      {
        label: <CheckBox />,
        key: 'select',
      },
      {
        label: 'Assigned To',
        key: 'assigned_to',
      },
      {
        label: 'Redemptions',
        key: 'redemptions',
      },
      {
        label: 'Code',
        key: 'title',
      },
      {
        label: 'Actions',
        key: 'actions',
      },
    ];

    this.state = {
      selectedFilter: null,
    };

    this.formatCouponData = this.formatCouponData.bind(this);
  }

  setSelectedFilter(selectedFilter) {
    this.setState({
      selectedFilter,
    });
  }

  getActionButton(code) {
    const { assigned_to: assignedTo, redemptions } = code;

    if (redemptions.used === redemptions.available) {
      return null;
    }

    const button = {
      label: 'Assign',
      onClick: () => {},
    };

    if (assignedTo) {
      button.label = 'Revoke';
      button.onClick = () => {};
    }

    return (
      <Button
        className={[
          'btn-link',
          'btn-sm',
          'pl-0',
        ]}
        label={button.label}
        onClick={button.onClick}
      />
    );
  }

  formatCouponData(data) {
    return data.map(code => ({
      ...code,
      redemptions: `${code.redemptions.used} of ${code.redemptions.available}`,
      actions: this.getActionButton(code),
      select: <CheckBox />,
    }));
  }

  render() {
    const { selectedFilter } = this.state;

    return (
      <div className="details row no-gutters px-2 my-3">
        <div className="col">
          <div className="details-header row mb-2">
            <div className="col-12 col-md-6 mb-2 mb-md-0">
              <H3>Coupon Details</H3>
            </div>
            <div className="col-12 col-md-6 mb-2 mb-md-0 text-md-right">
              <DownloadCsvButton id="coupon-details" fetchMethod={() => {}} />
            </div>
          </div>
          <div className="row mb-3">
            <div className="col d-flex justify-content-between align-items-center">
              <div className="filters d-flex align-items-center">
                <span className="mr-2">Filter:</span>
                <Button
                  className={['btn-sm']}
                  buttonType={selectedFilter === 'not-assigned' ? 'primary' : 'link'}
                  label="Not Assigned"
                  onClick={() => this.setSelectedFilter('not-assigned')}
                />
                <Button
                  className={['btn-sm']}
                  buttonType={selectedFilter === 'not-redeemed' ? 'primary' : 'link'}
                  label="Not Redeemed"
                  onClick={() => this.setSelectedFilter('not-redeemed')}
                />
                {selectedFilter &&
                  <Button
                    className={['btn-sm', 'btn-outline-primary', 'ml-2']}
                    label={
                      <React.Fragment>
                        <Icon className={['fa', 'fa-close', 'mr-1']} />
                        Clear filters
                      </React.Fragment>
                    }
                    onClick={() => this.setSelectedFilter(null)}
                  />
                }
              </div>
            </div>
          </div>
          <TableContainer
            id="coupon-details"
            className="coupon-details"
            fetchMethod={EcommerceApiService.fetchCouponDetails}
            columns={this.tableColumns}
            formatData={this.formatCouponData}
          />
        </div>
      </div>
    );
  }
}

export default CouponDetails;
