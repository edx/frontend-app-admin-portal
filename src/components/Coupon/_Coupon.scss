.coupon {
  box-shadow: 0 0.125rem 0.25rem rgba(0, 0, 0, 0.075);
  transition: opacity 0.2s ease;

  &.dimmed {
    opacity: 0.5;

    &:hover,
    &:focus {
      opacity: 1;
    }
  }

  &:not(.expanded):not(.border-danger) {
    &:hover,
    &:focus {
      border-color: $gray-300 !important; // `!important` needed to override .border Bootstrap class
    }
  }

  &.border-danger {
    $danger-100: map-get($theme-color-levels, 'danger-100');
    .metadata {
      background: $danger-100;

      &:hover,
      &:focus {
        background: darken($danger-100, 2%);
      }
    }
  }

  .metadata {
    transition: background 0.2s ease;
    position: relative;

    &:hover,
    &:focus {
      background: $gray-100;
      cursor: pointer;
    }

    .icons {
      position: absolute;
      top: 0;
      right: 0;

      @include media-breakpoint-up(lg) {
        position: relative;
      }
    }
  }

  &.expanded {
    &:hover,
    &:focus {
      background: inherit;
    }

    .metadata {
      background: #095686;
      border-radius: 0.1875rem 0.1875rem 0 0;
      color: #ffffff;
    }
  }
}
