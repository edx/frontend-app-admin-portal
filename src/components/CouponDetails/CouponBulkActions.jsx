import React from 'react';
import PropTypes from 'prop-types';

import {
  Button, Form,
} from '@edx/paragon';
import { BULK_ACTION } from './constants';

const CouponBulkActions = ({
  value, options, disabled, onChange, handleBulkAction,
}) => (
  <>
    <Form.Group className="flex-grow-1" controlId={BULK_ACTION.controlId} name={BULK_ACTION.name}>
      <Form.Control
        className="float-right"
        floatingLabel={BULK_ACTION.label}
        as="select"
        value={value}
        onChange={(e) => onChange(e.target.value)}
        disabled={disabled}
      >
        {options.map(
          ({ label, value: optionValue, disabled: optionDisabled }) => (
            <option
              key={optionValue}
              value={optionValue}
              disabled={optionDisabled}
            >
              {label}
            </option>
          ),
        )}
      </Form.Control>
    </Form.Group>
    <Button
      className="ml-2 mb-3"
      onClick={handleBulkAction}
      disabled={disabled}
    >
      Go
    </Button>
  </>
);

CouponBulkActions.propTypes = {
  value: PropTypes.string.isRequired,
  options: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired,
    disabled: PropTypes.bool.isRequired,
  })).isRequired,
  onChange: PropTypes.func.isRequired,
  disabled: PropTypes.bool.isRequired,
  handleBulkAction: PropTypes.func.isRequired,
};

export default CouponBulkActions;
