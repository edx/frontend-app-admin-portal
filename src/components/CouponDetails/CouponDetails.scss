@import "~bootstrap/scss/functions";
@import "~bootstrap/scss/variables";

.coupon-details {
    font-size: 1rem;

    .loading {
        font-size: inherit;
    }

    .alert {
        margin-bottom: 0;
        font-size: 0.85rem;

        .fa {
            font-size: 1rem;
        }
    }

    .table,
    .pagination .btn {
        font-size: 0.85rem;
    }

    .table {
        td {
            vertical-align: middle;
        }

        .btn {
            font-size: inherit;
            text-decoration: underline;

            &:focus {
                box-shadow: $btn-focus-box-shadow;
            }
        }
    }
    
    .toggles,
    .bulk-actions {
        .form-group {
            display: inline-block;

            &,
            label {
                margin: 0;
            }

            label {
                font-weight: 600;
            }
        }
    }
}
