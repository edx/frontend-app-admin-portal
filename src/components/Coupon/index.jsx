import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { Button, Icon } from '@edx/paragon';

import CouponDetails from './CouponDetails';

import { isTriggerKey } from '../../utils';

import './Coupon.scss';

const triggerKeys = {
  OPEN_DETAILS: [' ', 'Enter'],
  CLOSE_DETAILS: [' ', 'Enter', 'Escape'],
};

class Coupon extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      detailsExpanded: false,
      dimmed: false,
    };

    this.toggleCouponDetails = this.toggleCouponDetails.bind(this);
    this.handleCouponKeyDown = this.handleCouponKeyDown.bind(this);
  }

  setCouponOpacity(dimmedStatus) {
    this.setState({
      dimmed: dimmedStatus,
    });
  }

  closeCouponDetails() {
    this.setState({
      detailsExpanded: false,
      dimmed: false,
    });
  }

  toggleCouponDetails() {
    const detailsExpanded = !this.state.detailsExpanded;

    this.setState({
      detailsExpanded,
      dimmed: false,
    });

    if (detailsExpanded) {
      this.props.onExpand();
    } else {
      this.props.onCollapse();
    }
  }

  handleCouponKeyDown(event) {
    const { detailsExpanded } = this.state;
    if (!detailsExpanded && isTriggerKey({ triggerKeys, action: 'OPEN_DETAILS', key: event.key })) {
      event.preventDefault();
      this.toggleCouponDetails();
    } else if (detailsExpanded && isTriggerKey({ triggerKeys, action: 'CLOSE_DETAILS', key: event.key })) {
      event.preventDefault();
      this.toggleCouponDetails();
    }
  }

  renderExpandCollapseIcon() {
    const { detailsExpanded } = this.state;
    const iconClass = detailsExpanded ? 'fa-chevron-up' : 'fa-chevron-down';
    const screenReaderText = detailsExpanded ? 'Close' : 'Open';
    return (
      <Button
        className={[classNames(
          'toggle-details-btn',
          'btn-sm',
          {
            'text-white': detailsExpanded,
          },
        )]}
        buttonType="link"
        label={
          <Icon
            className={['fa', iconClass]}
            screenReaderText={`${screenReaderText} coupon details`}
          />
        }
        tabIndex="-1"
      />
    );
  }

  render() {
    const { detailsExpanded, dimmed } = this.state;
    const { data } = this.props;

    return (
      <div
        className={classNames(
          'coupon mb-3 mb-lg-2 rounded border',
          {
            expanded: detailsExpanded,
            dimmed,
          },
        )}
      >
        <div
          className="metadata row no-gutters p-2 d-flex align-items-center"
          onClick={this.toggleCouponDetails}
          onKeyDown={this.handleCouponKeyDown}
          role="button"
          tabIndex="0"
        >
          <div className="col-sm-12 col-lg-3 mb-2 mb-lg-0">
            <small className={classNames({ 'text-muted': !detailsExpanded, 'text-light': detailsExpanded })}>Coupon Name</small>
            <div>{data.title}</div>
          </div>
          <div className="col-sm-12 col-lg-4 mb-2 mb-lg-0">
            <div className="row no-gutters">
              <div className="col">
                <small className={classNames({ 'text-muted': !detailsExpanded, 'text-light': detailsExpanded })}>Valid From</small>
                <div>{data.validFromDate}</div>
              </div>
              <div className="col">
                <small className={classNames({ 'text-muted': !detailsExpanded, 'text-light': detailsExpanded })}>Valid To</small>
                <div>{data.validToDate}</div>
              </div>
            </div>
          </div>
          <div className="col-sm-12 col-lg-4 mb-2 mb-lg-0">
            <div className="row no-gutters">
              <div className="col">
                <small className={classNames({ 'text-muted': !detailsExpanded, 'text-light': detailsExpanded })}>Unassigned Codes</small>
                <div>{data.unassignedCodes}</div>
              </div>
              <div className="col">
                <small className={classNames({ 'text-muted': !detailsExpanded, 'text-light': detailsExpanded })}>Enrollments Redeemed</small>
                <div>
                  {`${data.enrollmentsRedeemed} of ${data.totalEnrollments}`}
                  <span className="ml-1">
                    {`(${Math.round((data.enrollmentsRedeemed / data.totalEnrollments) * 100)}%)`}
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div className="col-lg-1 order-first order-lg-last text-right">
            {this.renderExpandCollapseIcon()}
          </div>
        </div>
        {detailsExpanded && <CouponDetails />}
      </div>
    );
  }
}

Coupon.defaultProps = {
  onExpand: () => {},
  onCollapse: () => {},
};

Coupon.propTypes = {
  data: PropTypes.shape({
    // ...
  }).isRequired,
  onExpand: PropTypes.func,
  onCollapse: PropTypes.func,
};

export default Coupon;
