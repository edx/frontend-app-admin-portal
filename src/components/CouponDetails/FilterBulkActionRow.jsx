import React from 'react';
import PropTypes from 'prop-types';

import CouponFilters from './CouponFilters';
import CouponBulkActions from './CouponBulkActions';
import { COUPON_FILTER_TYPES } from './constants';

// Display component to handle layout

const FilterBulkActionRow = ({ couponFilterProps, couponBulkActionProps, ...sharedProps }) => (
  <div className="d-flex justify-content-between mt-3">
    <div className="toggles">
      <CouponFilters
        {...couponFilterProps}
        {...sharedProps}
      />
    </div>
    <div className="bulk-actions m-md-0 d-flex justify-content-end">
      <CouponBulkActions
        {...couponBulkActionProps}
        {...sharedProps}
      />
    </div>
  </div>
);

FilterBulkActionRow.propTypes = {
  // specific PropTypes are defined on the CouponFilters and CouponBulkActions components, respectively
  couponFilterProps: PropTypes.shape().isRequired,
  couponBulkActionProps: PropTypes.shape().isRequired,
  isTableLoading: PropTypes.bool.isRequired,
  selectedToggle: PropTypes.oneOf(Object.values(COUPON_FILTER_TYPES)).isRequired,

};

export default FilterBulkActionRow;
