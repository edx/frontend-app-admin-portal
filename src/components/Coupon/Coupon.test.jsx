import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';

import Coupon from './index';
import CouponDetails from '../CouponDetails';

const mockStore = configureMockStore([thunk]);
const store = mockStore({
  table: {
    'coupon-details': {
      loading: false,
      results: [],
    },
  },
  csv: {
    'coupon-details': {},
  },
});

const initialCouponData = {
  id: 1,
  title: 'Test Coupon',
  start_date: '2018-06-01T12:00:00Z',
  end_date: '2019-02-01T12:00:00Z',
  has_error: false,
  num_unassigned: 0,
  num_uses: 0,
  max_uses: 1,
};

const CouponWrapper = props => (
  <MemoryRouter>
    <Provider store={store}>
      <Coupon
        data={initialCouponData}
        {...props}
      />
    </Provider>
  </MemoryRouter>
);

describe('<Coupon />', () => {
  let wrapper;

  const simulateCouponClick = () => {
    wrapper.find(Coupon).find('.metadata').simulate('click');
  };

  const simulateCouponKeyDown = (key) => {
    wrapper.find(Coupon).find('.metadata').simulate('keydown', { key });
  };

  describe('renders correctly', () => {
    it('with max uses', () => {
      const tree = renderer
        .create((
          <CouponWrapper />
        ))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('without max uses', () => {
      const tree = renderer
        .create((
          <CouponWrapper
            data={{
              ...initialCouponData,
              max_uses: null,
            }}
          />
        ))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('with error state', () => {
      const tree = renderer
        .create((
          <CouponWrapper
            data={{
              ...initialCouponData,
              has_error: true,
            }}
          />
        ))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('expands and collapses correctly', () => {
    it('expands on click of collapsed coupon', () => {
      const mockOnExpand = jest.fn();
      wrapper = mount((
        <CouponWrapper
          onExpand={mockOnExpand}
        />
      ));

      simulateCouponClick();
      expect(mockOnExpand).toBeCalledTimes(1);
      expect(wrapper.find(Coupon).find(CouponDetails).prop('isExpanded')).toBeTruthy();
    });

    it('collapses on click of expanded coupon', () => {
      const mockOnCollapse = jest.fn();
      wrapper = mount((
        <CouponWrapper
          onCollapse={mockOnCollapse}
        />
      ));

      simulateCouponClick();
      expect(wrapper.find(Coupon).find(CouponDetails).prop('isExpanded')).toBeTruthy();

      simulateCouponClick();
      expect(mockOnCollapse).toBeCalledTimes(1);
      expect(wrapper.find(Coupon).find(CouponDetails).prop('isExpanded')).toBeFalsy();
    });

    it('expands on keydown of collapsed coupon', () => {
      const mockOnExpand = jest.fn();
      wrapper = mount((
        <CouponWrapper
          onExpand={mockOnExpand}
        />
      ));

      simulateCouponKeyDown('Enter');
      expect(mockOnExpand).toBeCalledTimes(1);
      expect(wrapper.find(Coupon).find(CouponDetails).prop('isExpanded')).toBeTruthy();
    });

    it('collapses on keydown of expanded coupon', () => {
      const mockOnCollapse = jest.fn();
      wrapper = mount((
        <CouponWrapper
          onCollapse={mockOnCollapse}
        />
      ));

      simulateCouponKeyDown('Enter');
      expect(wrapper.find(Coupon).find(CouponDetails).prop('isExpanded')).toBeTruthy();

      simulateCouponKeyDown('Enter');
      expect(mockOnCollapse).toBeCalledTimes(1);
      expect(wrapper.find(Coupon).find(CouponDetails).prop('isExpanded')).toBeFalsy();
    });

    it('does not handle unknown keydown event', () => {
      const mockOnExpandOrOnCollapse = jest.fn();
      const mockOnCollapse = jest.fn();
      wrapper = mount((
        <CouponWrapper
          onCollapse={mockOnExpandOrOnCollapse}
          onExpand={mockOnExpandOrOnCollapse}
        />
      ));

      simulateCouponKeyDown('A');

      expect(mockOnCollapse).toBeCalledTimes(0);
    });
  });

  it('sets state correctly on setCouponOpacity()', () => {
    wrapper = shallow(<CouponWrapper />);
    const instance = wrapper.find(Coupon).dive().instance();

    expect(instance.state.dimmed).toBeFalsy();
    instance.setCouponOpacity(true);
    expect(instance.state.dimmed).toBeTruthy();
  });

  it('sets state correctly on closeCouponDetails()', () => {
    wrapper = shallow(<CouponWrapper />);
    const instance = wrapper.find(Coupon).dive().instance();

    instance.setState({
      isExpanded: true,
    });

    expect(instance.state.isExpanded).toBeTruthy();
    instance.closeCouponDetails(true);
    expect(instance.state.isExpanded).toBeFalsy();
  });
});
