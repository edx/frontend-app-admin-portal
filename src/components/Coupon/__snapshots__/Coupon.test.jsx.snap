// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Coupon /> renders correctly with error state 1`] = `
<div
  className="coupon mb-3 mb-lg-2 rounded border border-danger"
>
  <div
    aria-controls="coupon-details-1"
    aria-expanded={false}
    className="metadata row no-gutters p-2 d-flex align-items-center rounded"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex="0"
  >
    <div
      className="col-sm-12 col-lg-3 mb-2 mb-lg-0"
    >
      <small
        className="text-muted"
      >
        Coupon Name
      </small>
      <div>
        Test Coupon
      </div>
    </div>
    <div
      className="col-sm-12 col-lg-4 mb-2 mb-lg-0"
    >
      <div
        className="row no-gutters"
      >
        <div
          className="col"
        >
          <small
            className="text-muted"
          >
            Valid From
          </small>
          <div>
            June 1, 2018
          </div>
        </div>
        <div
          className="col"
        >
          <small
            className="text-muted"
          >
            Valid To
          </small>
          <div>
            February 1, 2019
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-sm-12 col-lg-4 mb-2 mb-lg-0"
    >
      <div
        className="row no-gutters"
      >
        <div
          className="col"
        >
          <small
            className="text-muted"
          >
            Assignments Remaining
          </small>
          <div>
            0
          </div>
        </div>
        <div
          className="col"
        >
          <small
            className="text-muted"
          >
            Enrollments Redeemed
          </small>
          <div>
            0 of 1
            <span
              className="ml-1"
            >
              (0%)
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="icons col-lg-1 order-first order-lg-last text-right pr-2 mt-1 m-lg-0"
    >
      <span
        aria-hidden={true}
        className="fa fa-exclamation-circle text-danger mr-2"
        id="Icon3"
      />
      <span
        className="sr-only"
      >
        Coupon has an error.
      </span>
      <span
        aria-hidden={true}
        className="fa fa-chevron-down text-primary"
        id="Icon4"
      />
      <span
        className="sr-only"
      >
        Open coupon details
      </span>
    </div>
  </div>
  <div
    className="coupon-details row no-gutters px-2 my-3 d-none"
    id="coupon-details-1"
  >
    <div
      className="col"
    />
  </div>
</div>
`;

exports[`<Coupon /> renders correctly with max uses 1`] = `
<div
  className="coupon mb-3 mb-lg-2 rounded border"
>
  <div
    aria-controls="coupon-details-1"
    aria-expanded={false}
    className="metadata row no-gutters p-2 d-flex align-items-center rounded"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex="0"
  >
    <div
      className="col-sm-12 col-lg-3 mb-2 mb-lg-0"
    >
      <small
        className="text-muted"
      >
        Coupon Name
      </small>
      <div>
        Test Coupon
      </div>
    </div>
    <div
      className="col-sm-12 col-lg-4 mb-2 mb-lg-0"
    >
      <div
        className="row no-gutters"
      >
        <div
          className="col"
        >
          <small
            className="text-muted"
          >
            Valid From
          </small>
          <div>
            June 1, 2018
          </div>
        </div>
        <div
          className="col"
        >
          <small
            className="text-muted"
          >
            Valid To
          </small>
          <div>
            February 1, 2019
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-sm-12 col-lg-4 mb-2 mb-lg-0"
    >
      <div
        className="row no-gutters"
      >
        <div
          className="col"
        >
          <small
            className="text-muted"
          >
            Assignments Remaining
          </small>
          <div>
            0
          </div>
        </div>
        <div
          className="col"
        >
          <small
            className="text-muted"
          >
            Enrollments Redeemed
          </small>
          <div>
            0 of 1
            <span
              className="ml-1"
            >
              (0%)
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="icons col-lg-1 order-first order-lg-last text-right pr-2 mt-1 m-lg-0"
    >
      <span
        aria-hidden={true}
        className="fa fa-chevron-down text-primary"
        id="Icon1"
      />
      <span
        className="sr-only"
      >
        Open coupon details
      </span>
    </div>
  </div>
  <div
    className="coupon-details row no-gutters px-2 my-3 d-none"
    id="coupon-details-1"
  >
    <div
      className="col"
    />
  </div>
</div>
`;

exports[`<Coupon /> renders correctly without max uses 1`] = `
<div
  className="coupon mb-3 mb-lg-2 rounded border"
>
  <div
    aria-controls="coupon-details-1"
    aria-expanded={false}
    className="metadata row no-gutters p-2 d-flex align-items-center rounded"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex="0"
  >
    <div
      className="col-sm-12 col-lg-3 mb-2 mb-lg-0"
    >
      <small
        className="text-muted"
      >
        Coupon Name
      </small>
      <div>
        Test Coupon
      </div>
    </div>
    <div
      className="col-sm-12 col-lg-4 mb-2 mb-lg-0"
    >
      <div
        className="row no-gutters"
      >
        <div
          className="col"
        >
          <small
            className="text-muted"
          >
            Valid From
          </small>
          <div>
            June 1, 2018
          </div>
        </div>
        <div
          className="col"
        >
          <small
            className="text-muted"
          >
            Valid To
          </small>
          <div>
            February 1, 2019
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-sm-12 col-lg-4 mb-2 mb-lg-0"
    >
      <div
        className="row no-gutters"
      >
        <div
          className="col"
        >
          <small
            className="text-muted"
          >
            Assignments Remaining
          </small>
          <div>
            0
          </div>
        </div>
        <div
          className="col"
        >
          <small
            className="text-muted"
          >
            Enrollments Redeemed
          </small>
          <div>
            0
          </div>
        </div>
      </div>
    </div>
    <div
      className="icons col-lg-1 order-first order-lg-last text-right pr-2 mt-1 m-lg-0"
    >
      <span
        aria-hidden={true}
        className="fa fa-chevron-down text-primary"
        id="Icon2"
      />
      <span
        className="sr-only"
      >
        Open coupon details
      </span>
    </div>
  </div>
  <div
    className="coupon-details row no-gutters px-2 my-3 d-none"
    id="coupon-details-1"
  >
    <div
      className="col"
    />
  </div>
</div>
`;
